crates/codex/src/ttype/builder.rs

```diff
@@ -604,6 +605,51 @@ fn get_shape_from_ast(
         keyed_array.non_empty = shape.has_non_optional_fields() || shape.kind.is_non_empty();
 
         Ok(TAtomic::Array(TArray::Keyed(keyed_array)))
+    } else {
+        let mut shaped_object = TShapedObject::new();
+
+        shaped_object.known_items = Some({
+            let mut tree = BTreeMap::new();
+            let mut next_offset = 0;
+
+            for field in &shape.fields {
+                let field_is_optional = field.is_optional();
+
+                let key = match field.key.as_ref() {
+                    Some(field_key) => {
+                        let key = match field_key.key {
+                            ShapeKey::String { value, .. } => ArrayKey::String(atom(value)),
+                            ShapeKey::Integer { value, .. } => ArrayKey::Integer(value),
```

Comment:
we should error here. objects can't have integer property names.

We should probably have a new ast for `object{}`, because the name of the property can't contain `-` for example, which is currently allowed because we allow it for arrays.